tests/test_cases/test_cocotb/test_generator_coroutines.py

```diff
@@ -50,7 +50,7 @@ def test_function_not_decorated(dut):
     except TypeError as exc:
         assert "yielded" in str(exc)
         assert "scheduler can't handle" in str(exc)
-    except:
+    except Exception:  # FIXME: can we be more specific?
```

Comment:
I think this is actually a bug - translating from one execution to another here isn't really useful.

What makes this a bug is the fact it's not failing if no exception is thrown, which would be spelt:
```suggestion
    else:
```

Reply:
Done.